libshaderc_spvc/src/spvc.cc

```diff
@@ -209,26 +211,35 @@ shaderc_spvc_compilation_result_t validate_and_compile(
   }
 
   std::vector<uint32_t> intermediate_source;
-  if (options->transform_to_webgpu) {
-    spvtools::Optimizer opt(SPV_ENV_WEBGPU_0);
+  if (options->webgpu_to_vulkan) {
+    if (options->source_env != SPV_ENV_WEBGPU_0) {
+      result->messages.append(
+          "WARNING: Converting from WebGPU to Vulkan, with "
+          "non-WebGPU source env set, this may cause "
+          "unexpected behaviour...\n");
+    }
+    spvtools::Optimizer opt(SPV_ENV_VULKAN_1_1);
```

Comment:
It's TBD whether we go with Vulkan 1.1 semantics.  Off the top of my head, the main difference might be details of the data layout in buffers; but that's still not pinned down by WebGPU.  This is fine.